### ISSUE TITLE
Unnecessary elements in modal break styles

### ISSUE THREAD

**Issue by nitedani** [mantinedev/mantine#3854]
### What package has an issue

@mantine/core

### Describe the bug

I expected Modal to only render what's truly necessary, but it inserts divs that break my layout.
For example, the indicated div in the picture is unnecessary and breaks styles inside the Modal.Container.

![image](https://user-images.githubusercontent.com/67418286/226675656-fedda896-2a45-44a6-b9f0-aa67c7f89550.png)

### What version of @mantine/hooks page do you have in package.json?

6.0.4

### If possible, please include a link to a codesandbox with the reproduced problem

https://codesandbox.io/s/brave-stallman-6be3lj?file=/src/App.tsx

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

### Possible fix

Make Modal render only the minimal number of elements required. Remove the indicated div.
Ideally when I see one element in the code, I would like to see one element in the DOM. This is an exaggeration, but you get the point.
I expected `scrollAreaComponent={({ children }) => <>{children}</>}` to be the default. Then if I wanted something extra for scroll, I would have looked it up in the Modal docs page.

**Comment by dezoito** [comment]
FWIW I'm having a similar issue, where opening a modal seems to increase the margin at the top of my main container (thus shifting the content downwards).

Here's the relevant packages:

```js
  "dependencies": {
    "@emotion/react": "^11.10.6",
    "@mantine/core": "^6.0.0",
    "@mantine/form": "^6.0.0",
    "@mantine/hooks": "^6.0.0",
    "@mantine/notifications": "^6.0.0",
    "@mantine/prism": "^6.0.0",
    ...
```

**Comment by mattaningram** [comment]
Same problem here. New modal setup makes it surprisingly hard to do a basic drawer/modal with scrollable body but not scrollable header/footer. Exact same functionality is achievable with way fewer nested elements.

**Comment by rtivital** [comment]
Fixed in 6.0.5
